test/microbench/org/apache/cassandra/test/microbench/instance/WriteTest.java

```diff
@@ -0,0 +1,250 @@
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.commitlog.CommitLog;
+import org.apache.cassandra.db.memtable.Memtable;
+import org.apache.cassandra.utils.FBUtilities;
+import org.openjdk.jmh.annotations.*;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 10, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 1)
+@Threads(1)
+@State(Scope.Benchmark)
+public class WriteTest extends CQLTester
+{
+    static String keyspace;
+    String table;
+    ColumnFamilyStore cfs;
+    Random rand;
+
+    @Param({"1000"})
+    int BATCH = 1_000;
+
+    public enum EndOp
+    {
+        INMEM, TRUNCATE, FLUSH
+    }
+
+    @Param({"1000000"})
+    int count = 1_000_000;
+
+    @Param({"INMEM", "TRUNCATE", "FLUSH"})
+    EndOp flush = EndOp.INMEM;
+
+    @Param({"default"})
+    String memtableClass = "default";
+
+    @Param({"false"})
+    boolean useNet = false;
+
+    String writeStatement;
+
+    @Param({"32"})
+    int threadCount;
+
+    ExecutorService executorService;
+
+    @Setup(Level.Trial)
+    public void setup() throws Throwable
+    {
+        rand = new Random(1);
+        executorService = Executors.newFixedThreadPool(threadCount);
+        CQLTester.setUpClass();
+        CQLTester.prepareServer();
+        DatabaseDescriptor.setAutoSnapshot(false);
+        System.err.println("setupClass done.");
+        String memtableSetup = "";
+        if (!memtableClass.isEmpty())
+            memtableSetup = String.format(" AND memtable = '%s'", memtableClass);
+        keyspace = createKeyspace(
+        "CREATE KEYSPACE %s with replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 } and durable_writes = false");
+        table = createTable(keyspace,
+                            "CREATE TABLE %s ( userid bigint, picid bigint, commentid bigint, PRIMARY KEY(userid, picid)) with compression = {'enabled': false}" +
+                            memtableSetup);
+        execute("use " + keyspace + ";");
+        if (useNet)
+        {
+            CQLTester.requireNetwork();
+            executeNet(getDefaultVersion(), "use " + keyspace + ";");
+        }
+        writeStatement = "INSERT INTO " + table + "(userid,picid,commentid)VALUES(?,?,?)";
+        System.err.println("Prepared, batch " + BATCH + " threads " + threadCount + " flush " + flush);
+        System.err.println("Disk access mode " + DatabaseDescriptor.getDiskAccessMode() +
+                           " index " + DatabaseDescriptor.getIndexAccessMode());
+
+        cfs = Keyspace.open(keyspace).getColumnFamilyStore(table);
+        cfs.disableAutoCompaction();
+        cfs.forceBlockingFlush(ColumnFamilyStore.FlushReason.USER_FORCED);
+    }
+
+    @Benchmark
+    public void writeTable() throws Throwable
```

Comment:
Is the idea here to do as many of these write/flush cycles as we can do in 1 second for each iteration? `@Measurement` seems to have support for batching, so just curious. Would it make sense to benchmark the writes and flushes separately?